WorkloadManagementSystem/Client/Matcher.py

```diff
@@ -68,11 +68,11 @@ def selectJob( self, resourceDescription, credDict ):
     result = self.tqDB.matchAndGetJob( resourceDict, negativeCond = negativeCond )
 
     if not result['OK']:
-      return result
+      raise RuntimeError( result['Message'] )
     result = result['Value']
     if not result['matchFound']:
       self.log.info( "No match found" )
-      raise RuntimeError( "No match found" )
+      return {}
```

Comment:
The reason for using exceptions is for not having to return S_ERROR (and this case not even S_OK) because when I did it there was no DError.

The result of the matcher service for this specific case is an S_ERROR already, look at the changes I made in MatcherHandler.py